helper/resource/plugin.go

```diff
@@ -113,6 +114,58 @@ type providerFactories struct {
 	protov6 protov6ProviderFactories
 }
 
+func runProviderCommandApply(ctx context.Context, t testing.T, wd *plugintest.WorkingDir, factories *providerFactories) error {
```

Comment:
This is a light refactoring: these convenience functions remove incidental details from the main code flow.

Reply:
Love it!